aexpect/ops.py

```diff
@@ -0,0 +1,479 @@
+    :returns: standard output of stat(1) on ``path``
+    :rtype: str
+    """
+    if fmt is None:
+        return session.cmd_output(f"stat '{path}'")
+
+    if not isinstance(fmt, str) or fmt[0] != "%":
+        raise RuntimeError(f"{fmt} is not a valid format string for stat(1)")
+
+    return session.cmd_output(f"stat -c {fmt} '{path}'")
+
+
+def get_atime(session, path, human_readable=False):
+    """
+    Query the access time.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str path: path to the filesystem entry to stat
+    :param bool human_readable: whether to use human readable format or seconds
+                                since Epoch
+    :returns: standard output of stat(1) querying the access time of ``path``
+    :rtype: str
+    """
+    return stat(session, path, fmt=r"%x" if human_readable else "%X")
+
+
+def get_mtime(session, path, human_readable=False):
+    """
+    Query the modification time.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str path: path to the filesystem entry to stat
+    :param bool human_readable: whether to use human readable format or seconds
+                                since Epoch
+    :returns: standard output of stat(1) querying the last modification time
+              of ``path``
+    :rtype: str
+    """
+    return stat(session, path, fmt=r"%z" if human_readable else r"%Z")
+
+
+def get_ctime(session, path, human_readable=False):
+    """
+    Query the change time.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str path: path to the filesystem entry to stat
+    :param bool human_readable: whether to use human readable format or seconds
+                                since Epoch
+    :returns: standard output of stat(1) querying the last change time of
+              ``path``
+    :rtype: str
+    """
+    return stat(session, path, fmt=r"%y" if human_readable else r"%Y")
+
+
+def get_size(session, path):
+    """
+    Query the file size.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str path: path to the filesystem entry to stat
+    :returns: standard output of stat(1) querying the size of ``path``
+    :rtype: int
+    """
+    return int(stat(session, path, fmt=r"%s"))
+
+
+###############################################################################
+# test(1)
+###############################################################################
+
+
+def test(session, path, flags):
+    """
+    Wrapper for ``test``.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str path: path to the filesystem entry to test
+    :param str flags: flags passed to test
+    :returns: status of test(1) on ``path`` as True or False
+    :rtype: bool
+    """
+    return session.cmd_status(f"test {flags} '{path}'") == 0
```

Comment:
The `path` should use quotation.

Reply:
IIUC you mean to use `shlex.quote` here, I have also added it to a few more places.

Reply:
Done, thanks for adding more consistency observations